tests/integration/integration_suite_test.go

```diff
@@ -101,7 +108,16 @@ func TestFullStackScenario(t *testing.T) {
 		})
 
 		t.Run("Test Stargate", func(t *testing.T) {
-			testStargate(t, namespace)
+			// The backup/restore test runs before this. Because it shuts down
+			// the Cassandra cluster, we need to restart Stargate. See
+			// https://github.com/k8ssandra/k8ssandra/issues/411 for details.
+			releaseName := "k8ssandra"
+			dcName := "dc1"
+			if err := RestartStargate(releaseName, dcName, namespace); err == nil {
```

Comment:
What would you think of rather restarting Stargate at the end of the restore operation than here? Seems to make more sense since the issue is created by the restore.

Reply:
We may want to actually make that happen (if possible) at the end of full restores if we can (if that's possible), outside of the testing context.

Reply:
>What would you think of rather restarting Stargate at the end of the restore operation than here? Seems to make more sense since the issue is created by the restore.

It is a good point. I could go either way to be honest. As the test suite evolves this could be a problem elsewhere in the test code base until the issue is resolved in Stargate. But if the only restarts we have to worry about are with the restore, then the problem is somewhat isolated. So 🤷‍♂️ 

> We may want to actually make that happen (if possible) at the end of full restores if we can (if that's possible), outside of the testing context

Agreed, and this is something we could easily solve with a k8ssandra operator. It's not something I am particularly interested in solving with the current Helm based implementation.